idl/lpc55-pins.idol

```diff
@@ -0,0 +1,61 @@
+Interface(
+    name: "Pins",
+    ops: {
+	"iocon_configure_raw": (
+		args : {
+			"pin": ( type : "Pin", recv: FromPrimitive("u32")),
+			"conf": "u32",
+		},
+		reply: Result(
+			ok: "()",
+			err: CLike("GpioError")
+		),
+	),
+	"set_dir": (
+		args: {
+			"pin": ( type : "Pin", recv: FromPrimitive("u32")),
+			"dir": ( type: "Direction", recv: FromPrimitive("u32")),
+		},
+		reply: Result(
+			ok: "()",
+			err: CLike("GpioError"),
+		),
+	),
+
+
+
+	"set_val": (
+		args: {
+			"pin": ( type : "Pin", recv: FromPrimitive("u32")),
+			"val": ( type : "Value", recv: FromPrimitive("u8")),
+		},
+		reply: Result(
+			ok: "()",
+			err: CLike("GpioError"),
+		),
+	),
+
+
+
+	"read_val": (
+		encoding: Ssmarshal,
+		args: {
+			"pin": ( type : "Pin", recv: FromPrimitive("u32")),
+		},
+		reply: Result(
+			ok: "Value",
```

Comment:
Doing this brings the `hiffy` task from 23328 bytes to 14944 (!!), so it seems worthwhile!

(I didn't test that it works, though, so that would be good to check 😅)